src/hooks/useUploadsManager/useUploadsManager.tsx

```diff
@@ -69,34 +81,154 @@ export const UploadManagerProvider: React.FC = ({ children }) => {
     uploaded: {},
   })
 
-  // Will set all incomplete assets' status to missing on initial mount
   const isInitialMount = useRef(true)
   useEffect(() => {
+    if (!activeChannelId) {
+      return
+    }
     if (!isInitialMount.current) {
       return
     }
     isInitialMount.current = false
 
-    let missingAssetsCount = 0
-    uploadsState.forEach((asset) => {
-      if (asset.lastStatus !== 'completed') {
-        updateAsset(asset.contentId, 'missing')
-        missingAssetsCount++
-      } else {
-        setIgnoredAssetsIds((ignored) => [...ignored, asset.contentId])
+    const init = async () => {
+      const videosPromise = client.query<GetVideosQuery, GetVideosQueryVariables>({
+        query: GetVideosDocument,
+        variables: { where: { channelId_eq: activeChannelId } },
+      })
+      const channelPromise = client.query<GetChannelQuery, GetChannelQueryVariables>({
+        query: GetChannelDocument,
+        variables: { where: { id: activeChannelId } },
+      })
+
+      const [videosResponse, channelResponse] = await Promise.all([videosPromise, channelPromise])
+
+      const videos = videosResponse.data.videos
+      const channel = channelResponse.data.channelByUniqueInput
+
+      const videosLookup = Object.values(createLookup(videos || []))
+
+      // Adding missing video assets to uploadsState
+      videosLookup.forEach((video) => {
+        const isMediaPending = video.mediaAvailability === AssetAvailability.Pending
+        const isThumbnailPending = video.thumbnailPhotoAvailability === AssetAvailability.Pending
+
+        const isMediaPresentInUploads = uploadsState.some(
+          (asset) => asset.contentId === video.mediaDataObject?.joystreamContentId
+        )
+        const isThumbnailPresentInUploads = uploadsState.some(
+          (asset) => asset.contentId === video.thumbnailPhotoDataObject?.joystreamContentId
+        )
+
+        if (isMediaPending && !isMediaPresentInUploads) {
+          addAsset({
```

Comment:
I'm wondering, what's the point of adding that all to the local storage?

The approach we take here currently is that we use local storage as the source of truth - that's what we will be using to display the data later on. And we do a sync at the beginning to make it updated. But what if we took a different approach and decided that the data from the query node is the source of truth and whatever we have in the local storage can be an addition to that?

Reply:
Does that make sense? I think in that case we wouldn't need to add those assets to the state at all